numalogic/udfs/tools.py

```diff
@@ -273,7 +273,7 @@ def __fetch_ts(self, uuid: str, key: str) -> _DedupMetadata:
                 msg_train_records=_msg_train_records,
             )
 
-    def ack_insufficient_data(self, key: KEYS, uuid: str, train_records: int) -> bool:
+    def ack_insufficient_data(self, logger, key: KEYS, uuid: str, train_records: int) -> bool:
```

Comment:
can we have the typing? 
Also, can we have that Optional parameter? What do you think?

Reply:
I don't think logger should come in as an argument. Can you figure out some other way @gulshan02 ?

Reply:
done